frontend/components/admin/UserLists.tsx

```diff
@@ -0,0 +1,76 @@
+import { Box, useDisclosure, Spinner, SimpleGrid } from '@chakra-ui/react';
+import { Fragment, useState } from 'react';
+import UserList from './UserList';
+import UserInfoSimple from '@frontend/models/UserInfoSimple';
+import UserModal from './UserModal';
+import useRole from '@frontend/hooks/userRole';
+
+const UserLists = () => {
+  const { userRoles, isLoading, isError } = useRole(); // Custom Hook to Fetch the user/role data
+
+  const { isOpen, onOpen, onClose } = useDisclosure(); // Hook to deal with the modal visibility
+
+  const [userSelected, setUserSelected] = useState<UserInfoSimple>({
+    AccountId: 0,
+    FirstName: '',
+    LastName: '',
+    Role: '',
+  });
```

Comment:
Match the correct naming convention as mentionned in other comments for the back-end files, refer to the signUpParameters interface in the minicap/frontend/functions/signUp.ts file